client/components/mma/switch/SwitchContainer.tsx

```diff
@@ -60,24 +46,66 @@ export const SwitchContainer = () => {
 	const routerState = location.state as SwitchRouterState;
 	const productDetail = routerState?.productDetail;
 
+	if (!productDetail) {
+		return <AsyncLoadedSwitchContainer />;
+	}
+
+	return <RenderedPage productDetail={productDetail} />;
+};
+
+const AsyncLoadedSwitchContainer = () => {
+	const request = createProductDetailFetcher(
+		PRODUCT_TYPES.contributions.allProductsProductTypeFilterString,
+	);
+
+	const { data, loadingState } = useAsyncLoader<MembersDataApiResponse>(
+		request,
+		JsonResponseHandler,
+	);
+
+	if (loadingState == LoadingState.HasError) {
+		return (
+			<PageContainer
+				selectedNavItem={NAV_LINKS.accountOverview}
+				pageTitle={'Change your support'}
+				compactTitle
+			>
+				<GenericErrorScreen loggingMessage={false} />
+			</PageContainer>
+		);
+	}
+	if (loadingState == LoadingState.IsLoading) {
+		return (
+			<PageContainer
+				selectedNavItem={NAV_LINKS.accountOverview}
+				pageTitle={'Change your support'}
+				compactTitle
+			>
+				<DefaultLoadingView />
+			</PageContainer>
+		);
+	}
+
+	if (
+		data == null ||
+		data.products.length == 0 ||
+		data.products.filter(isProduct).length > 1
+	) {
+		return <Navigate to="/" />;
+	}
+
+	const productDetail = data.products.filter(isProduct)[0];
+	return <RenderedPage productDetail={productDetail} />;
+};
+
+const RenderedPage = (props: { productDetail: ProductDetail }) => {
 	return (
 		<PageContainer
 			selectedNavItem={NAV_LINKS.accountOverview}
 			pageTitle={'Change your support'}
 			compactTitle
 		>
-			{productDetail ? (
-				contextAndOutletContainer(productDetail, isFromApp())
-			) : (
-				<MembersDataApiAsyncLoader
-					fetch={createProductDetailFetcher(
-						PRODUCT_TYPES.contributions
-							.allProductsProductTypeFilterString,
-					)}
-					render={renderSingleProductOrReturnToAccountOverview}
-					loadingMessage={'Fetching your current support'}
-				/>
-			)}
+			{contextAndOutletContainer(props.productDetail, isFromApp())}
```

Comment:
`contextAndOutletContainer` isn't called anywhere else so couldn't the content just be included here directly?